tests/test_main.py

```diff
@@ -69,6 +72,36 @@ def test_extract_conda_pkgs(tmp_path: Path):
     run_conda("constructor", "--prefix", tmp_path, "--extract-conda-pkgs", check=True)
 
 
+def test_extract_tarball_umask(tmp_path: Path):
+    "Ported from https://github.com/conda/conda-package-streaming/pull/65"
+
+    def empty_tarfile_bytes(name, mode=0o644):
+        """
+        Return BytesIO containing a tarfile with one empty file named :name
+        """
+        tar = io.BytesIO()
+        t = tarfile.TarFile(mode="w", fileobj=tar)
+        tarinfo = tarfile.TarInfo(name=name)
+        tarinfo.mode = mode
+        t.addfile(tarinfo, io.BytesIO())
+        t.close()
+        tar.seek(0)
+        return tar
+
+    tarbytes = empty_tarfile_bytes(name="naughty_umask", mode=0o777)
+    process = subprocess.Popen(
+        [CONDA_EXE, "constructor", "--prefix", tmp_path, "--extract-tarball"],
+        stdin=subprocess.PIPE,
+        umask=0o022,
+    )
+    process.communicate(tarbytes.getvalue())
+    rc = process.wait()
+    assert rc == 0
+    if sys.platform != "win32":
+        mode = (tmp_path / "naughty_umask").stat().st_mode
+        assert not mode & stat.S_IWGRP, "%o" % mode
```

Comment:
This only checks group writing permissions. This is the origin of the bug, but shouldn't the entire umask be tested?

Reply:
Maybe. I copied the logic from Daniel's PR but maybe I can simply test the resulting chmod. Should be `0o777-0o22` right?

Reply:
New commit added, hopefully with the expected changes :)